### PR TITLE
slopShell

### PR DESCRIPTION
**Pull request by oldkingcone**
Used correct template. Added to groups blackarch, blackarch-backdoor. Let me know if you see any issues and I shall correct them.

### CODE REVIEW

File: packages/slopShell/PKGBUILD [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+pkgname=slopShell
+pkgver=1
+_SLOP_DIR="/opt/$pkgname-$pkgver/"
+pkgrel=1
+pkgdesc='PHP based backdoor/webshell with C2 capabilities'
+arch=('any')
+groups=('blackarch', 'blackarch-backdoor')
+url='https://github.com/oldkingcone/slopShell'
+license=('Apache License 2.0')
```
Comment (noraj) [inline packages/slopShell/PKGBUILD:12]: It's just Apache

- https://wiki.archlinux.org/title/PKGBUILD#license
- https://archlinux.org/packages/core/any/licenses/

File: packages/slopShell/PKGBUILD [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+depends=('postgresql-server', 'php', 'nginx', 'postgresql', 'php-pear', 'php-fpm')
+source=("https://github.com/oldkingcone/slopShell.git")
+# this includes .git directory
+full_sha512sums=('436AE54334F364C8ACC2F2ADF7B85EE2CD8C0DC462A7B17D75197AD382DBC19F')
```
Comment (noraj) [inline packages/slopShell/PKGBUILD:16]: this variable doesn't exist

File: packages/slopShell/PKGBUILD [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+# without .git directory
+sha512sums=('4A0F0EAADF1EEE376148847CC4C5B49FCFD494192C2F059F3BA5ECB499E2C41B')
+
+prepare() {
+if ! [ -e $_SLOP_DIR ];then
+    sudo git clone $url $_SLOP_DIR && sudo chown -R $( whoami ):postgres $_SLOP_DIR
```
Comment (noraj) [inline packages/slopShell/PKGBUILD:22]: use `cd $pkgname` else you'll clone two times

see template https://github.com/BlackArch/blackarch-pkgbuilds/blob/master/PKGBUILD-generic-git

File: packages/slopShell/PKGBUILD [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+pkgver=1
+_SLOP_DIR="/opt/$pkgname-$pkgver/"
+pkgrel=1
+pkgdesc='PHP based backdoor/webshell with C2 capabilities'
+arch=('any')
+groups=('blackarch', 'blackarch-backdoor')
+url='https://github.com/oldkingcone/slopShell'
+license=('Apache License 2.0')
+depends=('postgresql-server', 'php', 'nginx', 'postgresql', 'php-pear', 'php-fpm')
+source=("https://github.com/oldkingcone/slopShell.git")
```
Comment (noraj) [inline packages/slopShell/PKGBUILD:14]: `source=("git+https://github.com/foo/$pkgname.git")`

File: packages/slopShell/PKGBUILD [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+pkgname=slopShell
+pkgver=1
+_SLOP_DIR="/opt/$pkgname-$pkgver/"
+pkgrel=1
+pkgdesc='PHP based backdoor/webshell with C2 capabilities'
+arch=('any')
+groups=('blackarch', 'blackarch-backdoor')
+url='https://github.com/oldkingcone/slopShell'
+license=('Apache License 2.0')
+depends=('postgresql-server', 'php', 'nginx', 'postgresql', 'php-pear', 'php-fpm')
```
Comment (noraj) [inline packages/slopShell/PKGBUILD:13]: missing `makedepends=('git')`

**Review by noraj — CHANGES_REQUESTED** [review]

**Comment by noraj** [comment]
replace #3220

**Comment by oldkingcone** [comment]
first package, corrections applied.

File: packages/slopShell/PKGBUILD [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+    echo "Directory is already created."
+fi
+$( which pg_ctl ) initdb -D $_SLOP_DIR/slop_data -l $_SLOP_DIR/main.log
+cd $_SLOP_DIR/slop_data && $( which createdb ) sloppy_bots -E utf-8 -O $( whoami ) -h localhost -p 5432 -U postgres
+sudo $( which chmod ) 770 /var/run/postgresql
+sudo $( which chown ) postgres:postgres /var/run/postgresql
+$( which pg_ctl ) start -D $_SLOP_DIR/slop_data -l $_SLOP_DIR/main.log
+echo "If any of these commands failed, you will need to re-run them yourself."
+echo "Creating a setup script in your home directory."
+curl https://raw.githubusercontent.com/oldkingcone/slopShell/master/setup.sh -o ~/slop_setup.sh
```
Comment (ikstream) [inline packages/slopShell/PKGBUILD:34]: The user shouldn't run a script afterwards, to install the actual tool, but the PKGBUILD should take care of these steps

File: packages/slopShell/PKGBUILD (hunk 3 above)
Comment (ikstream) [reply packages/slopShell/PKGBUILD:23]: You don't need to clone, the build system will take care for you

**Comment by oldkingcone** [comment]
Did a little RTFM. Should be good now.

**Comment by noptrix** [comment]
@oldkingcone hey. thanks for your effort here. but a few comments here: we use tabwidth=2, don't use '4' please. use newlines. please stop using sys-tools inside PKGBUILD. better offer a `$tool.install` file. for style related things: simply copy a PKGBUILD-template or a PKGBUILD from a tool inside the repo and use that as a base. your `$pkgver` is not updated. you use static `1.0.0` but at the same time define `pkgver()` to create a version out of git HEAD. etc. etc. before continuing read the archwiki for `PKGBUILD` and `makepkg`. then take a look into our PKGBUILDs + templates.

**Comment by oldkingcone** [comment]
@noptrix the use of system tools, pg_ctl, createdb, chown and chmod are required for postgres. Ive noticed across multiple

**Pull request closed without merging**